thrift.go

```diff
@@ -0,0 +1,168 @@
+package baseplate
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/getsentry/raven-go"
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/reddit/baseplate.go/batcherror"
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+type baseplateThriftServer struct {
+	thriftServer *thrift.TSimpleServer
+	config       ServerConfig
+	closers      []io.Closer
+	logger       log.Wrapper
+	secretsStore *secrets.Store
+}
+
+func (bts *baseplateThriftServer) Config() ServerConfig {
+	return bts.config
+}
+func (bts *baseplateThriftServer) Secrets() *secrets.Store {
+	return bts.secretsStore
+}
+
+func (bts *baseplateThriftServer) Serve() error {
+	return bts.thriftServer.Serve()
+}
+
+func (bts *baseplateThriftServer) Close() error {
+	var errors batcherror.BatchError
+	errors.Add(bts.thriftServer.Stop())
+	for _, c := range bts.closers {
+		errors.Add(c.Close())
+	}
+	return errors.Compile()
+}
+
+func initLogger(cfg ServerConfig) log.Wrapper {
+	if cfg.Log.Level == "" {
+		cfg.Log.Level = log.InfoLevel
+	}
+	level := cfg.Log.Level
+	log.InitLogger(level)
+	return log.ZapWrapper(level)
```

Comment:
this function should return nothing. The log level defined in the config file is supposed to mean the minimal level of log to be seen (so info level means that all debug level logs are suppressed, while error level means that they don't want to see info level log as well). for all the other initialization that needs an `log.Wrapper`, just use `log.ZapWrapper(log.ErrorLevel)`.

Reply:
See also: https://github.com/reddit/baseplate.go/pull/153

Reply:
As stated above, I strongly disagree with having the bootstrap process being at a different log level than what the user explicitly asked for.

Reply:
The you need to provide two log level configs.

Reply:
See also: https://pkg.go.dev/github.com/reddit/baseplate.go/log?tab=doc#ErrorWithSentryWrapper

Reply:
This documentation is pretty unclear to me.  There's a little bit of the docs that is a direction of what to do, and not much how or why.

Reply:
I guess let's leave it for now because it's already encoded in the public API, but if I understand correctly, I would strongly like ErrorWithSentryWrapper to be deprecated in favor of something like a simple WithSentryWrapper.

Reply:
Forcing Error level logging isn't the right general purpose solution.  Bugs can happen in the bootstrap process and app authors should be able to adjust the log level.